### PR TITLE
Refactor sandbox selector component 

### PR DESCRIPTION
**Pull request by ninaceban**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.
- [ ] I've updated the schema in extension.json or no changes are necessary.
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Review by carterworks — APPROVED** [review]

File: src/view/components/sandboxSelector.jsx [hunk 1]
```diff
@@ -29,18 +29,22 @@ const getLabel = sandbox => {
 const SandboxSelector = ({
   defaultSelectedSandbox,
   onSelectionChange,
-  errorMessage,
-  initInfo
+  initInfo,
+  items,
```
Comment (jfkhoury) [inline src/view/components/sandboxSelector.jsx:33]: How about we rename these 2 options to what they actually are:

`items` -> `sandboxes`
`otherProps` -> `sandboxProps`

File: src/view/helpers/fetchSandboxes.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    if (e.name === "AbortError") {
+      throw e;
+    }
+
+    throw new UserReportableError(`Failed to load sandboxes.`, {
+      originatingError: e
+    });
+  }
+
+  return {
```
Comment (jfkhoury) [inline src/view/helpers/fetchSandboxes.js:40]: Can't we just return from within the `try`? And keep the `let...` inside the `try` as well?

**Review by jfkhoury — APPROVED** [review]
A couple of minor comments

File: src/view/components/sandboxSelector.jsx [hunk 3]
```diff
@@ -89,34 +93,36 @@ const SandboxSelector = ({
...
-      onSelectionChange={sandboxName => {
-        sandboxList.setSelectedKeys(new Set([sandboxName]));
-      }}
-      width="size-5000"
-    >
-      {item => {
-        return <Item key={item.name}>{getLabel(item)}</Item>;
-      }}
-    </Picker>
+    <>
```
Comment (jonsnyder) [inline src/view/components/sandboxSelector.jsx:96]: No need for the fragment here.

File: src/view/configuration/configuration.jsx [hunk 4]
```diff
@@ -138,7 +138,7 @@ const validationSchema = object().shape({
 
 const Configuration = ({ initInfo, context }) => {
   const [{ value: instances }] = useField("instances");
-  const [selectedTabKey, setSelectedTabKey] = useState(0);
+  const [selectedTabKey, setSelectedTabKey] = useState("0");
```
Comment (jonsnyder) [inline src/view/configuration/configuration.jsx:141]: This is the bug fix right?

**Review by jonsnyder — APPROVED** [review]
Do you want to hurry fix these now, or should we hold this for the next release and just include the fix?

File: src/view/configuration/configuration.jsx (hunk 4 above)
Comment (ninaceban) [reply src/view/configuration/configuration.jsx:141]: yes

**Review by carterworks — APPROVED** [review]

File: src/view/configuration/datastreamSelector.jsx [hunk 5]
```diff
@@ -148,61 +148,59 @@ const DatastreamSelector = ({
...
-              <Delete />
-            </ActionButton>
-            <Tooltip> Reset Datastream ID </Tooltip>
-          </TooltipTrigger>
-        </Flex>
+            isDisabled={!value}
+          >
+            <Delete />
+          </ActionButton>
+          <Tooltip> Reset Datastream ID </Tooltip>
```
Comment (jonsnyder) [inline src/view/configuration/datastreamSelector.jsx:200]: Lowercase "datastream" and add a period.

File: src/view/configuration/datastreamSelector.jsx [hunk 6]
```diff
@@ -148,61 +148,59 @@ const DatastreamSelector = ({
...
-            isDisabled={
-              !datastreamList.isLoading && !datastreamList.items.length
-            }
-            items={datastreamList.items}
           >
-            {item => {
-              return <Item key={getKey(item)}>{getLabel(item)}</Item>;
+            <Copy />
+          </ActionButton>
+          <Tooltip> Copy datastream ID to clipboard </Tooltip>
```
Comment (jonsnyder) [inline src/view/configuration/datastreamSelector.jsx:188]: Add a period.

File: test/functional/helpers/spectrum.js [hunk 7]
```diff
@@ -276,7 +276,12 @@ const componentWrappers = {
       },
       expectDisabled: createExpectDisabled(selector),
       expectEnabled: createExpectEnabled(selector.find("button")),
-      expectHidden: createExpectHidden(selector.parent().parent())
+      expectHidden: createExpectHidden(
+        selector
+          .parent()
+          .parent()
+          .parent()
+      )
```
Comment (jonsnyder) [inline test/functional/helpers/spectrum.js:284]: Can we make this more robust to react-spectrum changes? You can call parent with a selector string.

**Review by jonsnyder — APPROVED** [review]
This is great. I agree with Joe on returning inside the try.

File: test/functional/helpers/spectrum.js (hunk 7 above)
Comment (ninaceban) [reply test/functional/helpers/spectrum.js:284]: I looked into it but the selector is wrapped in multiple `div` with the same CSS classes

File: src/view/configuration/datastreamSelector.jsx (hunk 6 above)
Comment (ninaceban) [reply src/view/configuration/datastreamSelector.jsx]: thank you

**Pull request merged**